app/docs/md/enhance-ui/image.md

```diff
@@ -0,0 +1,240 @@
+- The browser will use whichever of these 3 generated images most closely fits the full width of the current viewport
+
+### Providing `sizes`
+```html
+<enhance-image
+  src="/_public/images/dog.jpg"
+  alt="My favourite dog"
+  sizes="(min-width: 48em) 50vw, 100vw"
+></enhance-image>
+```
+
+Again presuming the default configuration is being used in this example, the browser will select the generated image closest to 50% of the current viewport width when the viewport is 48em or wider. At viewports narrower than 48em, the image closest to the full width of the current viewport will be used.
+
+### Lazy loading and auto `sizes`
+```html
+<enhance-image
+  src="/_public/images/dog.jpg"
+  alt="My favourite dog"
+  loading="lazy"
+  sizes="auto"
+></enhance-image>
+```
+
+This is a good strategy to use for images that are rendered outside of the initial viewport area of a page. Since loading of the image will be deferred until the browser anticipates the image will be needed, the browser will already understand what size of image will be needed for the given content area, and can thus automatically determine which of your generated images to use.
+
+### Custom configuration
+
+```javascript
+// app/preflight.mjs
+export default async function Preflight ({ req }) {
+  return {
+    plugins: {
+      '@enhance/image': {
+        transform: {
+          widths: [1280, 1024, 720, 480, 375],
+          quality: 60,
+        }
+      }
+    }
+  }
+}
+```
+
+- Generated images will be available in widths of 1280px, 1024px, 720px, 480px, and 375px
+- Each generated image will be formatted at 60% quality
+- As no `format` option has been specified, the default format (webp) will be used
+
+```html
+<enhance-image
+  src="/_public/images/post-assets/dog.jpg"
+  alt="My favourite dog"
+  sizes="(min-width: 96em) 25vw, (min-width: 48em) 50vw, 100vw"
+></enhance-image>
+```
+
+- At viewports at least 96em wide, the generated image closest to `25vw` in width will be used; at viewports between 48–96em wide, the generated image closest to `50vw` in width will be used; at viewports narrower than 48em, the image closest to the full width of the current viewport will be used
+
+## Usage notes
+
+### `sizes` attribute
+
+The `sizes` attribute used by Enhance Image is simply a proxy for [`HTMLImageElement.sizes`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/sizes). This `sizes` attribute, as specified by [the HTML living standard](https://html.spec.whatwg.org/multipage/embedded-content.html#dom-img-sizes), can be tough to wrap your head around in certain cases. However, we believe it’s beneficial to remain close to the web platform, as it avoids the need to learn brittle abstractions and thus keeps knowledge reusable. As such, when you learn how to work with Enhance Image’s `sizes` attribute, you’re actually learning how to work with the Image element’s `sizes` attribute — warts and all.
```

Comment:
Love this breakdown. Well put!

Reply:
Thanks so much! It took quiiiite a while to walk this tightrope ;)